datachannel.go

```diff
@@ -104,19 +104,22 @@ func (api *API) newDataChannel(params *DataChannelParameters, log logging.Levele
 
 // open opens the datachannel over the sctp transport
 func (d *DataChannel) open(sctpTransport *SCTPTransport) error {
+	if sctpTransport == nil {
+		return errSCTPNotEstablished
+	}
+	sctpTransport.lock.RLock()
```

Comment:
@zyxar In a follow up commit `getAssocation` might be nice.  Wouldn't change the behavior of the code, but just nice to not use private members of other structs if we don't have too.

I have also thought about trying to move more stuff to internal. Will force us to write better code